### ISSUE TITLE
Test maven plugin

### ISSUE THREAD

**Issue by nulls** [saveourtool/diktat#1792]
Create a simple project on maven and configure maven plugin for diktat with version 2.0.0-rc.8

Todo:
- [x] Add example with multimodules
- [x] Test a single module
- [x] Test multi module
- [x] Test githubActions via `-Ddiktat.githubActions=true`
- [x] Test two reporters: `plain` configured via configuration and `githubActions` configured via environment variable ~(blocked by #1818)~
Test scenario:
1. Configure reporters:
```xml
<configuration>
    <diktatConfigFile>diktat-analysis.yml</diktatConfigFile>
    <inputs>
        <input>${project.basedir}/src/main/kotlin</input>
    </inputs>
    <reporters>
        <plain/>
        <json>
                <output>${project.built.directory}/some-folder/some-file.json</output>
        </json>
        <checkstyle/>
    </reporters>
</configuration>
```
2. Run diktat check and validate that diktat generates 3 reports: plain to stdout, json to some specific file and checkstyle (xml) to default location `target/reports/diktat/diktat.xml`
3. Run diktat check with environment variable `githubActions` and validate that diktat generates 4 reports: plain to stdout, json to some specific file, checkstyle (xml) to default location `target/reports/diktat/diktat.xml` and sarif report to default location `target/reports/diktat/diktat.sarif`
4. Add configuration for `gitHubActions` to `reporters`:
```xml
    <reporters>
        <gitHubActions/>
        <plain/>
        <json>
                <output>${project.built.directory}/some-folder/some-file.json</output>
        </json>
        <checkstyle/>
    </reporters>
```
And validate that diktat generates 4 reports with and without environment variable `githubActions`

Found issues:
- [x] #1814

**Comment by akuleshov7** [comment]
We already should have a simple maven project in our repo in examples

**Comment by nulls** [comment]
How to build snapshot version of plugin:
`gradlew.bat :diktat-api:publishToMavenLocal :diktat-common:publishToMavenLocal :diktat-rules:publishToMavenLocal :diktat-ktlint-engine:publishToMavenLocal :diktat-runner:publishToMavenLocal :diktat-maven-plugin:publishToMavenLocal`

This command builds and uploads a snapshot version to maven local repository. 
Version is printed in logs:
```
Reckoned version: 2.0.0-rc.8.113+6211a81
```
